packages/client/src/runtime/core/engines/data-proxy/utils/request.ts

```diff
@@ -40,11 +38,16 @@ export async function request(
   customFetch: (fetch: Fetch) => Fetch = (fetch) => fetch,
 ): Promise<RequestResponse> {
   const clientVersion = options.clientVersion
-  const jsRuntimeName = getJSRuntimeName()
 
   try {
-    if (jsRuntimeName === 'browser-like') {
-      return await customFetch(fetch)(url, options)
+    if (typeof fetch === 'function') {
+      const response = await customFetch(fetch)(url, options)
+
+      // needed to be able to call json() multiple times
+      response.json = () => response.clone().json()
+      response.text = () => response.clone().text()
```

Comment:
A "fetch" weirdness https://developer.mozilla.org/en-US/docs/Web/API/Response/clone

Reply:
iiuc, you cannot call `text` then `json`, vice-versa, or twice or more in a row hence the need for clone. The performance impact isn't clear. Perhaps I can try to find a better alternative that does not involve cloning, I will look into it again tomorrow.  Ideas welcome :)

Reply:
Why do we need to call `.json()` multiple times? (from the comment)

Reply:
+1 to the question. I'd rather fix us calling `json`/`text` multiple then introduce non-standard behaviour even to standard fetch.

Reply:
Calling `clone` can buffer the response into memory if the two streams are not read concurrently and at similar speeds. When one of them is not read, which would always be the case here for the original response, that one will buffer entirely into memory and (I think) stay until the GC can clean up the response object. I would also +1 locating the extra `json`/`text` calls and resolving it there if possible.

Reply:
Done